crates/chia-tools/src/bin/validate-blockchain-db.rs

```diff
@@ -0,0 +1,294 @@
+        let mut removals = HashSet::<[u8; 32]>::new();
+        while let Ok(Some(row)) = removals_rows.next() {
+            removals.insert(row.get::<_, [u8; 32]>(0).expect("missing coin_name"));
+        }
+        let mut additions_rows = select_created
+            .query([height])
+            .expect("failed to query created coins");
+        // coin-id -> (puzzle-hash, parent-coin, amount, reward)
+        let mut additions = HashMap::<[u8; 32], ([u8; 32], [u8; 32], u64, bool)>::new();
+        while let Ok(Some(row)) = additions_rows.next() {
+            let coin_name = row.get::<_, [u8; 32]>(0).expect("missing coin_name");
+            let reward = row.get::<_, bool>(1).expect("missing coinbase");
+            let ph = row.get::<_, [u8; 32]>(2).expect("missing puzzle_hash");
+            let parent = row.get::<_, [u8; 32]>(3).expect("missing parent");
+            let amount = u64::from_be_bytes(row.get::<_, [u8; 8]>(4).expect("missing amount"));
+            additions.insert(coin_name, (ph, parent, amount, reward));
+        }
+        pool.execute(move || {
+                let mut a = Allocator::new_limited(500_000_000);
+
+                let ti = block.transactions_info.as_ref().expect("transactions_info");
+                let generator = block
+                    .transactions_generator
+                    .as_ref()
+                    .expect("transactions_generator");
+
+                // after the hard fork, we run blocks without paying for the
+                // CLVM generator ROM
+                let block_runner = if height >= constants.hard_fork_height {
+                    run_block_generator2
+                } else {
+                    run_block_generator
+                };
+                let flags = ALLOW_BACKREFS
+                    | if args.skip_signature_validation {
+                        DONT_VALIDATE_SIGNATURE
+                    } else {
+                        0
+                    };
+                let conditions = block_runner(
+                    &mut a,
+                    generator,
+                    &block_refs,
+                    ti.cost,
+                    flags,
+                    &ti.aggregated_signature,
+                    None,
+                    constants,
+                )
+                .expect("failed to run block generator");
+
+                assert_eq!(conditions.cost, ti.cost);
+
+                for spend in &conditions.spends {
+                    let coin_name = *spend.coin_id;
+                    assert!(removals.remove(coin_name.as_slice()),
+                        "could not find coin {coin_name} in coin_record table, which is being spent at height {height}");
+                    for add in &spend.create_coin {
+                        let new_coin_id = Coin::new(coin_name, add.puzzle_hash, add.amount).coin_id();
+                        let Some(new_db_coin) = additions.get(new_coin_id.as_slice()) else {
```

Comment:
I would destructure this tuple if we're not going for a struct, to increase readability and avoid mistakes.

Reply:
```suggestion
                        let Some(&(new_db_coin_ph, new_db_coin_parent, new_db_coin_amount, new_db_coin_is_reward )) = additions.get(new_coin_id.as_slice()) else {
```